src/lib/array.jl

```diff
@@ -337,34 +337,6 @@ end
 # LinearAlgebra
 # =============
 
-@adjoint function transpose(x)
-  back(Δ) = (transpose(Δ),)
-  back(Δ::NamedTuple{(:parent,)}) = (Δ.parent,)
-  return transpose(x), back
-end
-
-@adjoint function LinearAlgebra.Transpose(x)
-  back(Δ) = (LinearAlgebra.Transpose(Δ),)
-  back(Δ::NamedTuple{(:parent,)}) = (Δ.parent,)
-  return LinearAlgebra.Transpose(x), back
-end
-
-
-@adjoint function Base.adjoint(x)
-  back(Δ) = (Δ',)
-  back(Δ::NamedTuple{(:parent,)}) = (Δ.parent,)
-  return x', back
-end
-
-@adjoint function LinearAlgebra.Adjoint(x)
-  back(Δ) = (LinearAlgebra.Adjoint(Δ),)
-  back(Δ::NamedTuple{(:parent,)}) = (Δ.parent,)
-  return LinearAlgebra.Adjoint(x), back
-end
-
-@adjoint parent(x::LinearAlgebra.Adjoint) = parent(x), ȳ -> (LinearAlgebra.Adjoint(ȳ),)
-@adjoint parent(x::LinearAlgebra.Transpose) = parent(x), ȳ -> (LinearAlgebra.Transpose(ȳ),)
```

Comment:
I forgot that  https://github.com/JuliaDiff/ChainRulesCore.jl/pull/446  isn't merged, so removing these lines probably isn't a good idea. 

Before:
```
julia> gradient(x -> first(parent(x)), [1,2,3]')[1]
1×3 adjoint(::Vector{Float64}) with eltype Float64:
 1.0  0.0  0.0
```
This PR, with tagged CRC:
```
julia> gradient(x -> first(parent(x)), [1,2,3]')[1]
(parent = [1.0, 0.0, 0.0],)
```

Reply:
One alternative idea is to desguar the `@adjoint` and overload `_pullback` directly. That'll have the benefit of removing some unnecessary layers, while making it easier to do stuff like https://github.com/FluxML/ZygoteRules.jl/issues/22.

Reply:
Glad to see there were tests for this, at least: fails on `isapprox(::NamedTuple{(:parent,), ...}, ::Matrix{Float64})`.